src/test/java/com/artipie/rpm/hm/NodeHasPkgCountTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.hm;
+
+import com.artipie.rpm.meta.XmlPackage;
+import com.jcabi.xml.XMLDocument;
+import java.io.FileNotFoundException;
+import java.nio.file.Paths;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsNot;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Matches;
+
+/**
+ * Test for {@link NodeHasPkgCount}.
+ * @since 0.10
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+class NodeHasPkgCountTest {
+
+    @Test
+    void countsPackages() throws FileNotFoundException {
+        MatcherAssert.assertThat(
+            new NodeHasPkgCount(2, XmlPackage.OTHER.tag()),
+            new Matches<>(
+                new XMLDocument(
+                    Paths.get("src/test/resources-binary/repodata/other.xml.example")
+                )
+            )
+        );
+    }
+
+    @Test
+    void doesNotMatchWhenPackagesAmountDiffers() throws FileNotFoundException {
+        MatcherAssert.assertThat(
+            new NodeHasPkgCount(10, XmlPackage.PRIMARY.tag()),
+            new IsNot<>(
+                new Matches<>(
```

Comment:
@olenagerasimova actually there is a dedicated Matcher called `Mismatches` for this, can you use it?

Reply:
@victornoel no, I cannot. `Mismatches` dedicated to check mismatch descriptions, I'm not testing description here.

Reply:
@olenagerasimova it also of course check what you are checking above, it's just that it is more complete so it also checks the description.

Reply:
@victornoel description is not properly done yet (I've added this part of the job in the todo)

Reply:
@olenagerasimova this shouldn't be a problem to use `Mismatches` I think. Anyway, let's forget about it then